pkg/controller/node.go

```diff
@@ -166,25 +168,30 @@ func NewNodeController(mgr manager.Manager, controllerSharedState *SharedState)
 			return true
 		},
 		UpdateFunc: func(e event.UpdateEvent) bool {
-			// make sure only update event from node taint changed from "unschedulable" gets enqueued to reconciler queue
+
 			old, oldOk := e.ObjectOld.(*corev1.Node)
 			new, newOk := e.ObjectNew.(*corev1.Node)
 
-			wasUnschedulable := false
-			nowSchedulable := true
-			for _, taint := range old.Spec.Taints {
-				if taint.Key == "node.kubernetes.io/unschedulable" {
-					wasUnschedulable = true
+			if oldOk && newOk {
+				// update event from node taint changed from "unschedulable" gets enqueued to reconciler queue
+				wasUnschedulable := false
+				nowSchedulable := true
+				for _, taint := range old.Spec.Taints {
+					if taint.Key == "node.kubernetes.io/unschedulable" {
+						wasUnschedulable = true
+					}
 				}
-			}
-			for _, taint := range new.Spec.Taints {
-				if taint.Key == "node.kubernetes.io/unschedulable" {
-					nowSchedulable = false
+				for _, taint := range new.Spec.Taints {
+					if taint.Key == "node.kubernetes.io/unschedulable" {
+						nowSchedulable = false
+					}
 				}
-			}
 
-			if oldOk && newOk && wasUnschedulable && nowSchedulable {
-				return true
+				// update event when LabelInstanceTypeStable is added to node
+				_, oldLableOk := old.GetLabels()[v1.LabelInstanceTypeStable]
+				_, newLabelOk := new.GetLabels()[v1.LabelInstanceTypeStable]
+
+				return (wasUnschedulable && nowSchedulable) || (!oldLableOk && newLabelOk)
```

Comment:
I am wondering if we could update the check to: 
return newLabelOk && ((wasUnschedulable && nowSchedulable) || !oldLableOk )

Reply:
For this case: `newLabelOk==true`, `wasUnschedulable==false`,  `nowSchedulable==true`, `oldLableOk==true`,
`return newLabelOk_true && ((wasUnschedulable_false && nowSchedulable_true) || !oldLableOk_!true ) `.  It will return false.
But it should return true, because nowSchedulable==true and newLabelOk==true which means we can retry attachments.

In addition, the readability is low. But I like the idea to keep most conditional checks in `predicate`.

Reply:
@alice-zheyan-yu for your case, I think we want it to return false since either schedulability or label doesn't change.

Reply:
@hccheng72 oh I see. Yes, you are right!